packages/sdk/src/evm/common/verification.ts

```diff
@@ -0,0 +1,324 @@
+import invariant from "tiny-invariant";
+import { ThirdwebSDK } from "../core";
+import { allChains } from "@thirdweb-dev/chains";
+import {
+  extractConstructorParamsFromAbi,
+  fetchSourceFilesFromMetadata,
+  resolveContractUriFromAddress,
+} from ".";
+import { ThirdwebStorage } from "@thirdweb-dev/storage";
+import { Abi } from "../schema";
+import { ethers, utils } from "ethers";
+import { getChainProvider } from "../constants";
+import { EtherscanResult } from "../types/verification";
+import fetch from "cross-fetch";
+
+//
+// ==================================
+// ======== Core Functions ==========
+// ==================================
+//
+
+export async function isVerifiedOnEtherscan(
+  contractAddress: string,
+  api: string,
+  apiKey: string,
+): Promise<boolean> {
+  const endpoint = `${api}?module=contract&action=getsourcecode&address=${contractAddress}&apikey=${apiKey}"`;
+
+  try {
+    const result = await fetch(endpoint, {
+      method: "GET",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json;charset=UTF-8",
+      },
+    });
+
+    const data = await result.json();
+    const etherscanResult = data.result[0] as EtherscanResult;
+    if (etherscanResult.ABI === "Contract source code not verified") {
+      return false;
+    }
+    return true;
+  } catch (e) {
+    throw new Error(
+      `Error checking verification for contract ${contractAddress}: ${e}`,
+    );
+  }
+}
+
+export async function verify(
+  contractAddress: string,
+  chainId: number,
+  api: string,
+  apiKey: string,
+  storage: ThirdwebStorage,
+) {
+  try {
+    const rpcUrl = allChains.find((chain) => chain.chainId === chainId)?.rpc[0];
```

Comment:
Instead of importing all chains here (which is way too much data) - how about passing a `Chain` object instead of chainId - which would have the RPC url inside already